File: frontend/src/api/system.ts
import request from './request'

export interface HwAccelSupport {
  supported: string[]
  available: Array<{ value: string; label: string }>
}

export default {
  /**
   * 获取系统支持的硬件加速列表
   */
  async getHwAccelSupport(): Promise<HwAccelSupport> {
    return request.get('/system/hw-accel')
  }
}

File: backend/app/api/system.py
"""系统信息 API"""
from fastapi import APIRouter
from app.services.hw_accel import get_supported_hw_accels

router = APIRouter(prefix="/system", tags=["system"])


@router.get("/hw-accel")
async def get_hw_accel_support():
    """获取系统支持的硬件加速列表"""
    return {
        "supported": get_supported_hw_accels(),
        "available": [
            {"value": "auto", "label": "自动"},
            {"value": "none", "label": "禁用"},
            {"value": "nvenc", "label": "NVIDIA NVENC"},
            {"value": "qsv", "label": "Intel QSV"},
            {"value": "vaapi", "label": "VAAPI"},
            {"value": "videotoolbox", "label": "VideoToolbox"},
            {"value": "amf", "label": "AMD AMF"}
        ]
    }
